### ISSUE TITLE
segm mAP remains zero!

### ISSUE THREAD

**Issue by farshid-mi** [open-mmlab/mmdetection#3336]
hi,consider mask_rcnn with **config below** and using a **custom dataset prepared in coco style(json)** files on **colab** env:

> model = dict(
>     type='MaskRCNN',
>     pretrained='torchvision://resnet50',
>     backbone=dict(
>         type='ResNet',
>         depth=50,
>         num_stages=4,
>         out_indices=(0, 1, 2, 3),
>         frozen_stages=0,
>         norm_cfg=dict(type='BN', requires_grad=True),
>         norm_eval=True,
>         style='pytorch'),
>     neck=dict(
>         type='FPN',
>         in_channels=[256, 512, 1024, 2048],
>         out_channels=256,
>         num_outs=5),
>     rpn_head=dict(
>         type='RPNHead',
>         in_channels=256,
>         feat_channels=256,
>         anchor_generator=dict(
>             type='AnchorGenerator',
>             scales=[8],
>             ratios=[0.5, 1.0, 2.0],
>             strides=[4, 8, 16, 32, 64]),
>         bbox_coder=dict(
>             type='DeltaXYWHBBoxCoder',
>             target_means=[0.0, 0.0, 0.0, 0.0],
>             target_stds=[1.0, 1.0, 1.0, 1.0]),
>         loss_cls=dict(
>             type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
>         loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
>     roi_head=dict(
>         type='StandardRoIHead',
>         bbox_roi_extractor=dict(
>             type='SingleRoIExtractor',
>             roi_layer=dict(type='RoIAlign', output_size=7, sampling_ratio=0),
>             out_channels=256,
>             featmap_strides=[4, 8, 16, 32]),
>         bbox_head=dict(
>             type='Shared2FCBBoxHead',
>             in_channels=256,
>             fc_out_channels=1024,
>             roi_feat_size=7,
>             num_classes=1,
>             bbox_coder=dict(
>                 type='DeltaXYWHBBoxCoder',
>                 target_means=[0.0, 0.0, 0.0, 0.0],
>                 target_stds=[0.1, 0.1, 0.2, 0.2]),
>             reg_class_agnostic=False,
>             loss_cls=dict(
>                 type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
>             loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
>         mask_roi_extractor=dict(
>             type='SingleRoIExtractor',
>             roi_layer=dict(type='RoIAlign', output_size=14, sampling_ratio=0),
>             out_channels=256,
>             featmap_strides=[4, 8, 16, 32]),
>         mask_head=dict(
>             type='FCNMaskHead',
>             num_convs=4,
>             in_channels=256,
>             conv_out_channels=256,
>             num_classes=1,
>             loss_mask=dict(
>                 type='CrossEntropyLoss', use_mask=True, loss_weight=1.0))))
> train_cfg = dict(
>     rpn=dict(
>         assigner=dict(
>             type='MaxIoUAssigner',
>             pos_iou_thr=0.7,
>             neg_iou_thr=0.3,
>             min_pos_iou=0.3,
>             match_low_quality=True,
>             ignore_iof_thr=-1),
>         sampler=dict(
>             type='RandomSampler',
>             num=256,
>             pos_fraction=0.5,
>             neg_pos_ub=-1,
>             add_gt_as_proposals=False),
>         allowed_border=-1,
>         pos_weight=-1,
>         debug=False),
>     rpn_proposal=dict(
>         nms_across_levels=False,
>         nms_pre=2000,
>         nms_post=1000,
>         max_num=1000,
>         nms_thr=0.7,
>         min_bbox_size=0),
>     rcnn=dict(
>         assigner=dict(
>             type='MaxIoUAssigner',
>             pos_iou_thr=0.5,
>             neg_iou_thr=0.5,
>             min_pos_iou=0.5,
>             match_low_quality=True,
>             ignore_iof_thr=-1),
>         sampler=dict(
>             type='RandomSampler',
>             num=512,
>             pos_fraction=0.25,
>             neg_pos_ub=-1,
>             add_gt_as_proposals=True),
>         mask_size=28,
>         pos_weight=-1,
>         debug=False))
> test_cfg = dict(
>     rpn=dict(
>         nms_across_levels=False,
>         nms_pre=1000,
>         nms_post=1000,
>         max_num=1000,
>         nms_thr=0.7,
>         min_bbox_size=0),
>     rcnn=dict(
>         score_thr=0.05,
>         nms=dict(type='nms', iou_threshold=0.5),
>         max_per_img=100,
>         mask_thr_binary=0.5))
> dataset_type = 'CocoDataset'
> classes = ('wheat', )
> data_root = ''
> img_norm_cfg = dict(
>     mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
> train_pipeline = [
>     dict(type='LoadImageFromFile'),
>     dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
>     dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
>     dict(type='RandomFlip', flip_ratio=0.5),
>     dict(
>         type='Normalize',
>         mean=[123.675, 116.28, 103.53],
>         std=[58.395, 57.12, 57.375],
>         to_rgb=True),
>     dict(type='Pad', size_divisor=32),
>     dict(type='DefaultFormatBundle'),
>     dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels', 'gt_masks'])
> ]
> test_pipeline = [
>     dict(type='LoadImageFromFile'),
>     dict(
>         type='MultiScaleFlipAug',
>         img_scale=(1333, 800),
>         flip=False,
>         transforms=[
>             dict(type='Resize', keep_ratio=True),
>             dict(type='RandomFlip'),
>             dict(
>                 type='Normalize',
>                 mean=[123.675, 116.28, 103.53],
>                 std=[58.395, 57.12, 57.375],
>                 to_rgb=True),
>             dict(type='Pad', size_divisor=32),
>             dict(type='ImageToTensor', keys=['img']),
>             dict(type='Collect', keys=['img'])
>         ])
> ]
> data = dict(
>     samples_per_gpu=4,
>     workers_per_gpu=0,
>     train=dict(
>         type='CocoDataset',
>         classes=('wheat', ),
>         ann_file='/content/data_xywh_pruned/train_ins.json',
>         img_prefix='',
>         pipeline=[
>             dict(type='LoadImageFromFile'),
>             dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
>             dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
>             dict(type='RandomFlip', flip_ratio=0.5),
>             dict(
>                 type='Normalize',
>                 mean=[123.675, 116.28, 103.53],
>                 std=[58.395, 57.12, 57.375],
>                 to_rgb=True),
>             dict(type='Pad', size_divisor=32),
>             dict(type='DefaultFormatBundle'),
>             dict(
>                 type='Collect',
>                 keys=['img', 'gt_bboxes', 'gt_labels', 'gt_masks'])
>         ]),
>     val=dict(
>         type='CocoDataset',
>         classes=('wheat', ),
>         ann_file='/content/data_xywh_pruned/val_ins.json',
>         img_prefix='',
>         pipeline=[
>             dict(type='LoadImageFromFile'),
>             dict(
>                 type='MultiScaleFlipAug',
>                 img_scale=(1333, 800),
>                 flip=False,
>                 transforms=[
>                     dict(type='Resize', keep_ratio=True),
>                     dict(type='RandomFlip'),
>                     dict(
>                         type='Normalize',
>                         mean=[123.675, 116.28, 103.53],
>                         std=[58.395, 57.12, 57.375],
>                         to_rgb=True),
>                     dict(type='Pad', size_divisor=32),
>                     dict(type='ImageToTensor', keys=['img']),
>                     dict(type='Collect', keys=['img'])
>                 ])
>         ]),
>     test=dict(
>         type='CocoDataset',
>         classes=('wheat', ),
>         ann_file='/content/data_xywh_pruned/test_ins.json',
>         img_prefix='',
>         pipeline=[
>             dict(type='LoadImageFromFile'),
>             dict(
>                 type='MultiScaleFlipAug',
>                 img_scale=(1333, 800),
>                 flip=False,
>                 transforms=[
>                     dict(type='Resize', keep_ratio=True),
>                     dict(type='RandomFlip'),
>                     dict(
>                         type='Normalize',
>                         mean=[123.675, 116.28, 103.53],
>                         std=[58.395, 57.12, 57.375],
>                         to_rgb=True),
>                     dict(type='Pad', size_divisor=32),
>                     dict(type='ImageToTensor', keys=['img']),
>                     dict(type='Collect', keys=['img'])
>                 ])
>         ]))
> evaluation = dict(metric=['bbox', 'segm'])
> optimizer = dict(type='SGD', lr=0.015, momentum=0.9, weight_decay=0.0001)
> optimizer_config = dict(grad_clip=None)
> lr_config = dict(
>     policy='step',
>     warmup='linear',
>     warmup_iters=500,
>     warmup_ratio=0.001,
>     step=[4])
> total_epochs = 5
> checkpoint_config = dict(interval=1)
> log_config = dict(
>     interval=50,
>     hooks=[dict(type='TextLoggerHook'),
>            dict(type='TensorboardLoggerHook')])
> dist_params = dict(backend='nccl')
> log_level = 'INFO'
> load_from = 'https://open-mmlab.s3.ap-northeast-2.amazonaws.com/mmdetection/v2.0/mask_rcnn/mask_rcnn_r50_fpn_1x_coco/mask_rcnn_r50_fpn_1x_coco_20200205-d4b0c5d6.pth'
> resume_from = None
> workflow = [('train', 1)]

during training model above i faced a confusing situation,while **acc**,**bbox_map_50** **increased** and loss_bbox,loss_cls,loss_mask,loss_rpn_bbox,loss_rpn_cls,**loss** all **decreased**, ****segm_map_50 remains almost zero**** the whole time(almost never shows anything but zero in evaluation table in train and test!)
also using inference_detector,it only **shows bboxes properly** with few stains at the right place but not the whole object!(not even for all instances)
what do u think is the problem?!is there anything that i should to add to dataset rather than segmentation(polygon) field?!is there anything that i should set in order to get beter (ins segmentation) in reference time?!
thank u in advance!

**Comment by ZwwWayne** [comment]
Please provide more information using the `Reimplementation Question`.

**Comment by farshid-mi** [comment]
more details provided here:
https://github.com/open-mmlab/mmdetection/issues/3338#issue-659282072 #
